bundles/org.openhab.binding.haassohnpelletoven/src/main/java/org/openhab/binding/haassohnpelletoven/internal/HaassohnpelletstoveHandler.java

```diff
@@ -0,0 +1,390 @@
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.haassohnpelletoven.communication.HaassohnpelletstoveJSONCommunication;
+import org.openhab.binding.haassohnpelletoven.data.HaassohnpelletstoveJsonData;
+import org.openhab.binding.haassohnpelletoven.helper.Helper;
+import org.openhab.binding.haassohnpelletoven.validation.IpAddressValidator;
+import org.openhab.binding.haassohnpelletoven.validation.PinValidator;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HaassohnpelletstoveHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Christian Feininger - Initial contribution
+ */
+@NonNullByDefault
+public class HaassohnpelletstoveHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(HaassohnpelletstoveHandler.class);
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private @Nullable HaassohnpelletstoveConfiguration config;
+    boolean resultOk = false;
+
+    private HaassohnpelletstoveJSONCommunication serviceCommunication;
+
+    private boolean automaticRefreshing = false;
+
+    private HashMap<String, Boolean> linkedChannels = new HashMap<String, Boolean>();
+
+    public HaassohnpelletstoveHandler(Thing thing) {
+        super(thing);
+        serviceCommunication = new HaassohnpelletstoveJSONCommunication();
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (channelUID.getId().equals(CHANNEL_prg)) {
+
+            String postData = null;
+            if (command.equals(OnOffType.ON)) {
+                postData = "{\"prg\":true}";
+            } else if (command.equals(OnOffType.OFF)) {
+                postData = "{\"prg\":false}";
+            }
+
+            if (postData != null) {
+                logger.debug("Executing {} command", CHANNEL_prg);
+                updateOvenData(postData);
+            }
+        } else if (channelUID.getId().equals(CHANNEL_spTemp)) {
+            if (command instanceof QuantityType<?>) {
+                QuantityType<?> value = null;
+                try {
+                    value = (QuantityType<?>) command;
+                    double a = value.doubleValue();
```

Comment:
Users can configure their temperature in Celsius and Fahrenheit in the UI. If a user configured Fahrenheit, the value returned by `.doubleValue()` will be in Fahrenheit. You need to convert it to Celsius. Otherwise the oven will be fed by the Fahrenheit value.

Reply:
ok